Survival-DEV/blog-backend
# Blog 31: User authentication

### :dart:  What is been done in this PR:
- [x] Add a JWT & Local guards authentication
- [x] Made sure that the password is been Hashed in the db
- [ ] Create tests ~ fix them
- [x] Fix the unAuthorized problem 

**Related to #31**

----
### :compass:   How to test it
Steps to test my PR:
1. `npm run ci`
2. make sure to setup your `env` file to have these params:
> DATABASE_URL=<your db url#>
> NODE_ENV=DEBUG  //for now
> logDB=["query", "error"]
> PORT=<your port#>
> TRUSTED_IP=["127.0.0.1"]
> WHITELIST=['http://localhost']
>ALLOWED_METHODS='GET,HEAD,PUT,PATCH,POST,DELETE,OPTIONS'
> JWT_SECRET=`<whatever string>`
> EXPIRESIN='60s'

4. `docker-compose up`   --to run redis 
3. `npm run start:dev`   --in another terminal
